sanssouci/lambda_calibration.py

```diff
@@ -88,6 +88,53 @@ def get_perm_p(X, categ, B=100, row_test_fun=stats.ttest_ind):
     return pval0
 
 
+def get_permuted_p_values_one_sample(X, B=100):
+    """
+    Get permutation p-values: Get a matrix of p-values under the null
+    hypothesis obtained by sign-flipping (one-sample test).
+
+    Parameters
+    ----------
+
+    X : array-like of shape (n,p)
+        numpy array of size [n,p], containing n observations of p variables
+        (hypotheses)
+    B : int
+        number of sign-flippings to be performed (default=100)
+
+    Returns
+    -------
+
+    pva0 : array-like of shape (B, p)
+        A numpy array of size [B,p], whose entry i,j corresponds to
+        p_{(j)}(g_i.X) with notation of the AoS 2020 paper cited below
+        (section 4.5) [1]_
+
+    References
+    ----------
+
+    .. [1] Blanchard, G., Neuvial, P., & Roquain, E. (2020). Post hoc
+        confidence bounds on false positives using reference families.
+        Annals of Statistics, 48(3), 1281-1303.
+    """
+
+    # Init
+    n, p = X.shape
+
+    # intialise p-values
+    pval0 = np.zeros([B, p])
+
+    for b in range(B):
+        X_flipped = (X.T * (2 * np.random.randint(-1, 1, size=n) + 1)).T
+        _, pval0[b] = stats.ttest_1samp(X_flipped, 0)
+
+    # Convert to p-values
```

Comment:
you can remove that one.

Reply:
thx, done in e4b7824538533c2400415a7844cfb5306f5c8149